Sources/bip32-ed25519-swift/bip32-ed25519-swift.swift

```diff
@@ -227,14 +231,16 @@ public class Bip32Ed25519 {
         let zr = z.subdata(in: ED25519_SCALAR_SIZE ..< 2 * ED25519_SCALAR_SIZE)
 
         // left = kl + 8 * trunc28(zl)
-        // right = zr + kr
-
+        // right = zr + kr mod 2^256
         let left = BigUInt(Data(kl.reversed())) + BigUInt(Data(trunc256MinusGBits(zl: zl, g: g).reversed())) * BigUInt(8)
+        guard left < (BigUInt(1) << 255) else {
```

Comment:
Why the bitshift? Check for being less than a unsigned number composed by all 1's?  Isn't there a clearer way of doing it?

Reply:
<img width="324" alt="Screenshot 2024-06-11 at 17 47 10" src="https://github.com/algorandfoundation/bip32-ed25519-swift/assets/83883690/7d700927-4160-4cc7-9468-954adce61bfd">

Reply:
Replaced it with the exponentiation